framework/source/class/qx/ui/table/cellrenderer/Boolean.js

```diff
@@ -131,12 +131,39 @@ qx.Class.define("qx.ui.table.cellrenderer.Boolean",
     // overridden
     _identifyImage : function(cellInfo)
     {
-      var imageHints =
+      var w;
+      var h;
+      var rm;
+      var id;
+      var ids;
+      var imageHints;
+
+      // Retrieve the ID
+      rm = qx.util.ResourceManager.getInstance();
+      ids = rm.getIds(this.__iconUrlTrue);
+
+      // If ID was found, we'll use its first (likely only) element here.
+      if (ids)
       {
-        imageWidth  : 11,
-        imageHeight : 11
+        id = ids[0];
+
+        // Get the natural size of the image
+        w = rm.getImageWidth(id);
+        h = rm.getImageHeight(id);
+      }
+
+      // Create the size portion of the hint.
+      //
+      // The traditional (fixed) size of the image was 11x11px. Use that if we
+      // weren't able to retrieve the actual size of the image, and never
+      // exceed that size.
+      imageHints =
+      {
+        imageWidth  : w ? Math.min(w, 11) : 11,
```

Comment:
this means there can never be a boolean image larger than 11 pixel in size ?

Reply:
It means that if the image size is larger than 11x11, we revert to using the prior behavior of specifying 11x11. I believe this is the best backward-compatibility solution. Previously, it was hard-coded at 11x11, so with this change, images smaller than 11x11 render correctly even with the addition of the #9610 changes.

Reply:
as a theme developer, this seems very wrong to me ... having icons forcefully rescaled to 11x11 pixel ... one shudders ... I would suggest du dop this whole thing and just check that our existing themes look ok .... apart from that, let people shoot themselves. silently resizing seems extremely wrong.

Reply:
But isn't @derrell argument that we just need to make sure things work as expected while accommodating @jpriebe's patch? As far as I understand the icon was force resized to 11x11 before, too. That might be a bad idea, but at least is B/C. Or did I misunderstand something?

Reply:
@cboulanger is exactly correct. The code is currently broken. Images smaller than 11x11 used to work fine, but are now being stretched and look awful. This code is trying to make things work like they did before, without introducing new, non-BC changes.

It is trivial to subclass the Boolean cell renderer and override that one method, if a different behavior is desired. Anyone who needed larger images previously would have already done that.

Reply:
@oetiker Would it be ok to accept this state for v6 and put the correction on the list of breaking changes for v7.0?

Reply:
sure as you wish ...